core/services/ocr2/delegate.go

```diff
@@ -586,7 +586,11 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.ServiceCtx, error) {
 		oracleCtx := job.NewServiceAdapter(oracles)
 		return []job.ServiceCtx{runResultSaver, vrfProvider, dkgProvider, oracleCtx}, nil
 	case job.OCR2Keeper:
-		keeperProvider, rgstry, encoder, logProvider, err2 := ocr2keeper.EVMDependencies(jb, d.db, lggr, d.chainSet, d.pipelineRunner)
+		credName, err2 := jb.OCR2OracleSpec.PluginConfig.MercuryCredentialName()
+		if err2 != nil {
+			return nil, errors.Wrap(err2, "failed to get mercury credential name")
+		}
+		keeperProvider, rgstry, encoder, logProvider, err2 := ocr2keeper.EVMDependencies(jb, d.db, lggr, d.chainSet, d.pipelineRunner, d.cfg.MercuryCredentials(credName))
```

Comment:
What happens now when the credential is missing?
Would it be more clear to have an explicit nil check and error here, instead of passing it along?

Reply:
after we check and confirm it's nil, we still need to pass along a nil right?
or you mean to make it explicit that we are passing down a nil?

Reply:
If passing nil is OK then this is fine. I thought that 'not found' was an error condition before.